### PR TITLE
feat(user-profile): implement actor profiles in user profile service

### PR DESCRIPTION
**Pull request by Herdismaria**
[Task](https://www.notion.so/Add-actor-profiles-table-endpoints-000d426492f343f3b5db58e39f810e40?pvs=4)
## What

Add new table for actor profiles in user profile service that stores email notification preferences for user's delegation.
Create endpoints to get, update and create actor profiles for users.

## Why

We want users to choose if they should receive emails intended for users that have granted them delegations.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Introduced new functionality to create or update actor profiles.
  - Added capability to retrieve actor profiles based on national ID.

- **Bug Fixes**
  - Replaced incorrect use of `Query` decorator with `Put` for updating profiles.

- **Documentation**
  - Removed outdated method `getDelegationPreferences` from documentation.

- **Tests**
  - Enhanced testing for actor profile fetching and updating functionalities.
  - New test cases added for handling actor profiles based on delegations.

- **Chores**
  - Updated service configurations and client imports for better service integration.

### CODE REVIEW

**Review by GunnlaugurG — APPROVED** [review]

**Review by GunnlaugurG — APPROVED** [review]
LGTM!

File: apps/services/user-profile/src/app/v2/me-user-profile.controller.ts [hunk 1]
```diff
@@ -138,4 +143,42 @@ export class MeUserProfileController {
...
     )
   }
+
+  @Get('/actor-profiles')
+  @Scopes(UserProfileScope.read)
+  @Documentation({
+    description: 'Get actor profiles for the current user.',
+    response: { status: 200, type: PaginatedActorProfileDto },
+  })
+  getDelegationPreferences(
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/me-user-profile.controller.ts:153]: Bot methods should add audit log. The get function can use the `@Audit` decorator, while the patch method should use the `AuditService`.

Also the get function should be renamed to `getActorProfiles`

File: apps/services/user-profile/src/app/v2/me-user-profile.controller.ts [hunk 2]
```diff
@@ -138,4 +143,42 @@ export class MeUserProfileController {
       this.userProfileService.confirmNudge(user.nationalId, input.nudgeType),
     )
   }
+
+  @Get('/actor-profiles')
+  @Scopes(UserProfileScope.read)
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/me-user-profile.controller.ts:148]: I'm wondering if we should limit company access to the actor profile get and patch.

Companies get the user profile read and write scope, but it doesn't make sense for companies (procuration holders) to check and manage these settings. So this screen won't be available for companies and we might wan't to limit the api access as well. We could just use the `ApiScope.internal` instead, as that is a scope that only signed in users will get, no delegation session will get it.

File: apps/services/user-profile/src/app/v2/models/actor-profile.model.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  })
+  @ApiProperty()
+  id!: string
+
+  @Column({
+    type: DataType.STRING,
+    allowNull: false,
+    unique: {
+      name: 'actor_profile_to_from_unique',
+      msg: 'combined unique constraint failed',
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/models/actor-profile.model.ts:36]: ~ Typo? Also below.

```suggestion
      msg: 'combined unique constraint field',
```

File: apps/services/user-profile/src/app/v2/test/me-user-profile.controller.spec.ts [hunk 4]
```diff
@@ -1119,4 +1122,249 @@ describe('MeUserProfileController', () => {
...
+          nationalId: testUserProfile.nationalId,
+          scope: [UserProfileScope.read],
+        }),
+      })
+
+      server = request(app.getHttpServer())
+      fixtureFactory = new FixtureFactory(app)
+      delegationsApi = app.get(DelegationsApi)
+      userProfileModel = app.get(getModelToken(UserProfile))
+      delegationPreferenceModel = app.get(getModelToken(ActorProfile))
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/test/me-user-profile.controller.spec.ts:1148]:
```suggestion
      actorProfileModel = app.get(getModelToken(ActorProfile))
```

File: apps/services/user-profile/src/app/v2/user-profile.controller.ts [hunk 5]
```diff
@@ -54,4 +65,38 @@ export class UserProfileController {
...
+    description: 'Get actor profiles for nationalId.',
+    request: {
+      header: {
+        'X-Param-From-National-Id': {
+          required: true,
+          description: 'National id of the user the .....',
+        },
+        'X-Param-To-National-Id': {
+          required: true,
+          description: 'National id of the user',
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/user-profile.controller.ts:80]: Update descriptions to match each param

File: apps/services/user-profile/src/app/v2/user-profile.controller.ts [hunk 6]
```diff
@@ -54,4 +65,38 @@ export class UserProfileController {
...
+        },
+      },
+    },
+    response: { status: 200, type: ActorProfileDto },
+  })
+  async getActorProfile(
```
Comment (saevarma) [inline apps/services/user-profile/src/app/v2/user-profile.controller.ts:86]: Add audit log

File: apps/services/user-profile/src/app/v2/me-user-profile.controller.ts (hunk 1 above)
Comment (Herdismaria) [reply apps/services/user-profile/src/app/v2/me-user-profile.controller.ts:153]: Forgot about that 🙈

File: apps/services/user-profile/src/app/v2/models/actor-profile.model.ts (hunk 3 above)
Comment (Herdismaria) [reply apps/services/user-profile/src/app/v2/models/actor-profile.model.ts:36]: I think this is an error message 🤔

**Pull request merged**